api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+              .put(
+                  BypassTimeTargetProperty.TWO_FACTOR_AUTH_BYPASS_TIME,
+                  AccessModuleName.TWO_FACTOR_AUTH)
+              .put(BypassTimeTargetProperty.RAS_LINK_LOGIN_GOV, AccessModuleName.RAS_LOGIN_GOV)
+              .put(
+                  BypassTimeTargetProperty.DATA_USE_AGREEMENT_BYPASS_TIME,
+                  AccessModuleName.DATA_USER_CODE_OF_CONDUCT)
+              .build();
+
+  @Autowired
+  public AccessModuleServiceImpl(
+      Provider<List<DbAccessModule>> dbAccessModulesProvider,
+      Clock clock,
+      UserAccessModuleDao userAccessModuleDao,
+      UserDao userDao,
+      UserServiceAuditor userServiceAuditor,
+      Provider<WorkbenchConfig> configProvider) {
+    this.dbAccessModulesProvider = dbAccessModulesProvider;
+    this.clock = clock;
+    this.userAccessModuleDao = userAccessModuleDao;
+    this.userDao = userDao;
+    this.userServiceAuditor = userServiceAuditor;
+    this.configProvider = configProvider;
+  }
+
+  @Override
+  public void updateBypassTime(long userId, AccessModule accessModuleName, boolean isBypassed) {
+    final DbUser user = userDao.findUserByUserId(userId);
+    final List<DbUserAccessModule> dbUserAccessModules = userAccessModuleDao.getAllByUser(user);
+    DbAccessModule accessModule =
+        dbAccessModulesProvider.get().stream()
+            .filter(a -> a.getName() == clientAccessModuleToStorage(accessModuleName))
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new BadRequestException(
+                        "There is no access module named: " + accessModuleName.toString()));
+    if (!accessModule.getBypassable()) {
+      throw new ForbiddenException("Bypass: " + accessModuleName.toString() + " is not allowed.");
+    }
+    Optional<DbUserAccessModule> retrievedAccessModule =
+        dbUserAccessModules.stream()
+            .filter(m -> m.getAccessModule().getName().equals(accessModule.getName()))
+            .findFirst();
+    final Timestamp newBypassTime;
+    final Timestamp previousBypassTime;
+    final DbUserAccessModule userAccessModuleToUpdate;
+
+    logger.info(
+        String.format(
+            "Bypassing %s(uid: %d) for module %s",
+            user.getContactEmail(), userId, accessModule.getName()));
+    previousBypassTime = retrievedAccessModule.map(DbUserAccessModule::getBypassTime).orElse(null);
+    newBypassTime = isBypassed ? new Timestamp(clock.instant().toEpochMilli()) : null;
+    userAccessModuleToUpdate =
+        retrievedAccessModule
+            .orElseGet(() -> new DbUserAccessModule().setUser(user).setAccessModule(accessModule))
+            .setBypassTime(newBypassTime);
+
+    userAccessModuleDao.save(userAccessModuleToUpdate);
+    if (configProvider.get().featureFlags.enableAccessModuleRewrite) {
+      // If enabled, fire audit event from here instead of from UserService.
+      userServiceAuditor.fireAdministrativeBypassTime(
```

Comment:
this should be a new "accessModuleServiceAuditor" or similar because this isn't the UserService

Reply:
and if we are firing different audit events then it's not a duplicate... and we don't need to introduce the feature flag yet.

Reply:
Ah, I thought `UserServiceAuditor` can host all user related changes, like `fireSetFreeTierDollarLimitOverride` not user services. 
I feel the UserService is different that the code's UserService. Like AccessModuleService is just a class make it cleaner and easier to writer test. But the actual change can actually happened inside the code's UserService. I can also just put it in UserService then like how we use `fireUpdateAccessTiersAction`  (that is not DbUser, but DbAccessTier).
(2) The feature flag is needed regardlessly by the UI and the following shouldUserBeRegistered change...

Reply:
OK, I see UserServiceAuditor doesn't work how I thought it did.  It would be good to think about this some more in the future, but what you have in this PR makes sense for how it works now.

I prefer the user access decision code to live here rather than UserService.  I would eventually like to move more of that code here, I think.

2 - OK